src/index.js

```diff
@@ -67,13 +67,11 @@ export default function factory({
     let i = 0;
     const len = handlers.length;
     const loop = async (next) => handlers[i++](req, res, next);
-    const next = (err) => {
-      i < len
-        ? err
-          ? onError(err, req, res, next)
-          : loop(next).catch(next)
-        : done && done(err);
-    };
+    const next = (err) => i < len
+      ? err
+        ? onError(err, req, res, next)
```

Comment:
What would be its behavior when an error is thrown. In this case the execution inside the handler still goes on?

Reply:
Yeah that's something I totally missed in my testing. my bad! Indeed errors aren't halting execution after `await next()`.

It seems it's behaving this way because the error is handled inside of `next()` and never returned back. I found a solution that ensures that errors are propagated, but I can't find one that gives `onError()` an option to call `next()`. Any ideas?

Reply:
We may need to decide on a behavior.

One thing we can do is to let `next()` throws and the error should propagate all the way up and only attach the error handler to the first handler:

```
handler1: await() -- onError()
     middleware1: await next() -- ⬆️
          middleware2: await next() - ⬆️
              middleware3: throw new Error()
```

But that is a breaking change since it will break if user have not been doing `await next()`, which is probably the case now.

Reply:
My pending changes resolves error handling without having to unravel the stack. In the event that the error handler wishes to ignore the error and call `next()`, however, it leads to the following unexpected behaviour.

```
// ❌ = does not execute
// Middleware listed after .get() represents code to be executed after await next().

// Example: GET /
middleware1: await next()
    middleware2: await next()
        middleware3: throw new Error() -> onError() -> next()
            .get(handler): 
        middleware3: ❌ -- not expected to run, it gave up control after throwing
    middleware2: ❌ -- expected to run
middleare1: ❌ -- expected to run
```

This is where I'm running into problems with finding a decent solution.